jcommune-view/jcommune-web-view/src/main/webapp/resources/javascript/app/topicDraft.js

```diff
@@ -0,0 +1,559 @@
+/*
+ * Copyright (C) 2011  JTalks.org Team
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+$(function () {
+    'use strict';
+
+    var SECOND = 1000,
+        MINUTE = 60 * SECOND,
+        HOUR = 60 * MINUTE;
+
+    var MIN_TITLE_LENGTH = 1,
+        MAX_TITLE_LENGTH = 120,
+
+        MIN_CONTENT_LENGTH = 1,
+        MAX_CONTENT_LENGTH = 20000,
+
+        MIN_POLL_TITLE_LENGTH = 1,
+        MAX_POLL_TITLE_LENGTH = 120,
+
+        MIN_POLL_ITEM_LENGTH = 1,
+        MAX_POLL_ITEM_LENGTH = 50,
+
+        MIN_ITEMS_NUMBER = 1,
+        MAX_ITEMS_NUMBER = 50;
+
+    var TITLE_ERROR_MESSAGE = $labelMessageSizeValidation
+            .replace('{min}', MIN_TITLE_LENGTH).replace('{max}', MAX_TITLE_LENGTH),
+
+        CONTENT_ERROR_MESSAGE = $labelMessageSizeValidation
+            .replace('{min}', MIN_CONTENT_LENGTH).replace('{max}', MAX_CONTENT_LENGTH),
+
+        POLL_TITLE_ERROR_MESSAGE = $labelMessageSizeValidation
+            .replace('{min}', MIN_POLL_TITLE_LENGTH).replace('{max}', MAX_POLL_TITLE_LENGTH),
+
+        POLL_ITEMS_VALUE_ERROR_MESSAGE = $pollItemsSize
+            .replace('{min}', MIN_ITEMS_NUMBER).replace('{max}', MAX_ITEMS_NUMBER),
+
+        POLL_ITEM_LENGTH_ERROR = $pollItemLength
+            .replace('{min}', MIN_POLL_ITEM_LENGTH).replace('{max}', MAX_POLL_ITEM_LENGTH);
+
+    /**
+     * Class for access to topic draft API
+     *
+     * @constructor
+     */
+    function TopicDraftApi() {
+        this._saveLock = false;
```

Comment:
Need to describe how these locks are used, what operations they are locking and why we need them. 

Also, isn't having one lock for both save and remove operations makes more sense? If not, please add comments why we need two of them separately.